File: process_test.js
const METHOD = process.argv.slice(2)[0];

function reader() {
  process.send({
    hello: "123"
  });
}

function writer() {
  process.on("message", function (msg, handle) {
    process.send(`10087`, handle);
  });
}

let methods = {
  reader: reader,
  writer: writer
};

methods[METHOD]();

File: ipc/process_test.go
package ipc

import (
	"os/exec"
	"strings"
	"testing"
)

func TestExec_Reader(t *testing.T) {
	cmd := exec.Command("node", "process_test.js", "reader")
	channel, err := Exec(cmd)
	if err != nil {
		t.Fatal(err)
	}
	defer func() {
		cmd.Process.Kill()
	}()

	msg := <-channel.Reader
	const expectedContent = "{\"hello\":\"123\"}\n"
	if strings.Compare(string(msg.Data), expectedContent) != 0 {
		t.Fatal("Message not matched: ", string(msg.Data))
	}
}
